pmm-app/tests/verifyRemoteInstances_test.js

```diff
@@ -113,3 +113,66 @@ Scenario(
     assert.ok(response.data.data.result.length !== 0, `Custom Metrics Should be available but got empty ${result}`);
   },
 );
+
+Scenario(
+  'PMM-T637 - Verify elements on HAProxy page @not-pr-pipeline',
+  async ({ I, remoteInstancesPage }) => {
+    I.amOnPage(remoteInstancesPage.url);
+    remoteInstancesPage.waitUntilRemoteInstancesPageLoaded();
+    remoteInstancesPage.openAddRemotePage('haproxy');
+    I.waitForVisible(remoteInstancesPage.fields.returnToMenuButton, 30);
+    I.waitForVisible(remoteInstancesPage.fields.hostName, 30);
+    I.waitForVisible(remoteInstancesPage.fields.serviceName, 30);
+    I.waitForVisible(remoteInstancesPage.fields.portNumber, 30);
+    I.waitForVisible(remoteInstancesPage.fields.userName, 30);
+    I.waitForVisible(remoteInstancesPage.fields.password, 30);
+    I.waitForVisible(remoteInstancesPage.fields.environment, 30);
+    I.waitForVisible(remoteInstancesPage.fields.region, 30);
+    I.waitForVisible(remoteInstancesPage.fields.availabilityZone, 30);
+    I.waitForVisible(remoteInstancesPage.fields.replicationSet, 30);
+    I.waitForVisible(remoteInstancesPage.fields.cluster, 30);
+    I.waitForVisible(remoteInstancesPage.fields.customLabels, 30);
+    I.waitForVisible(remoteInstancesPage.fields.skipConnectionCheck, 30);
+  },
+);
+
+Scenario(
+  'PMM-T636 - Verify adding HAProxy with all empty fields @not-pr-pipeline',
+  async ({ I, remoteInstancesPage }) => {
+    I.amOnPage(remoteInstancesPage.url);
+    remoteInstancesPage.waitUntilRemoteInstancesPageLoaded();
+    remoteInstancesPage.openAddRemotePage('haproxy');
+    I.waitForVisible(remoteInstancesPage.fields.addService, 30);
+    I.click(remoteInstancesPage.fields.addService);
+    I.waitForVisible(remoteInstancesPage.fields.requiredFieldHostname, 30);
+  },
+);
+
+Scenario(
+  'PMM-T635 - Verify Adding HAProxy service via UI @not-pr-pipeline',
+  async ({
+    I, remoteInstancesPage, pmmInventoryPage,
+  }) => {
+    const serviceName = 'haproxy_remote';
+    const url = process.env.PMM_UI_URL;
+
+    I.amOnPage(remoteInstancesPage.url);
+    remoteInstancesPage.waitUntilRemoteInstancesPageLoaded();
+    remoteInstancesPage.openAddRemotePage('haproxy');
+    I.waitForVisible(remoteInstancesPage.fields.hostName, 30);
+    I.fillField(remoteInstancesPage.fields.hostName, url.slice(7, url.length - 1));
```

Comment:
Maybe it's better to use [`URL`](https://developer.mozilla.org/ru/docs/Web/API/URL/URL) here, and use `url.host` instead ?

Reply:
I don't understand it. If you mean `url.slice(7, url.length - 1)` change to `url.host` so that's doesn't work.

Reply:
It doesn't work because in your case `url` is not of type `URL`.
First you should create it like this: 
```suggestion
    const url = new URL(stringUrl);
```

and then use `url.host`

Reply:
Thanks! :) Great, that's working! :)